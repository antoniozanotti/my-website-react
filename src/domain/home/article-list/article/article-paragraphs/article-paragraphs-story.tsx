import { ArticleParagraphs } from "./article-paragraphs";
import { ArticleParagraphsStoryInterface } from "./article-paragraphs-story-interface";

export function ArticleParagraphsStory(props: ArticleParagraphsStoryInterface) {
  return (
    <ArticleParagraphs
      paragraphs={props.paragraphs}
      showParagraph={() => true}
      showSeeMore={() => true}
      setIsOpen={(isOpenParam) => props.setIsOpen(isOpenParam)}
    />
  );
}
