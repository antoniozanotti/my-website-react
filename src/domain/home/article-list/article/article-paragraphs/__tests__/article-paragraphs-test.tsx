import {
  render,
  firstComponent,
  screen,
  fireEvent,
  getAllByText,
} from "@/lib/test";
import { ArticleParagraphsStory } from "./article-paragraphs-story";

describe("ArticleParagraphs", () => {
  it("should render paragraphs and handle callbacks", async () => {
    let isOpen = false;
    const { container } = render(
      <ArticleParagraphsStory
        paragraphs={["Paragraph1", "Paragraph2"]}
        setIsOpen={(isOpenParam) => {
          isOpen = isOpenParam;
        }}
      />
    );
    const component = firstComponent(screen);
    const paragraphs = screen.getAllByRole("paragraph");
    const seeMore = getAllByText(container, "See more...")[0];

    fireEvent.click(seeMore);

    expect(component).toBeVisible();
    expect(paragraphs).toHaveLength(2);
    expect(component).toHaveTextContent("Paragraph1");
    expect(component).toHaveTextContent("Paragraph2");
    expect(isOpen).toBeTruthy();
  });
});
