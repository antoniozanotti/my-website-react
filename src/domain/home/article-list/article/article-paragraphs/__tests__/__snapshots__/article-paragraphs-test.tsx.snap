// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleParagraphs should render paragraphs and handle callbacks 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <div
          class="space-y-4"
        >
          <p
            style="opacity: 0;"
          >
            Paragraph1
             
            <span
              class="underline text-accent-1 dark:text-dark-accent-1 hover:text-secondary hover:dark:text-dark-secondary"
            >
              See more...
            </span>
          </p>
          <p
            style="opacity: 0;"
          >
            Paragraph2
             
            <span
              class="underline text-accent-1 dark:text-dark-accent-1 hover:text-secondary hover:dark:text-dark-secondary"
            >
              See more...
            </span>
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
